### DISCUSSION TITLE
Website is extremely laggy on Safari desktop browser

### DISCUSSION THREAD

**Discussion by ScheevRapid** [matteobruni/tsparticles#1988] [Q&A] [answered]
Greetings,

I integrated tsParticles on my website and It is working wonderfully, except on desktop safari. Website is extremely laggy there. Is there a way to fix this? If not, is there some way to disable it only on safari? 

Any help appreciated.

Website: https://qualtec.sk/

**Comment by matteobruni** [comment, marked as answer]
Hi @ScheevRapid,

My main browser is Safari, I tried also your website but I haven't noticed any lag there. My MacBook Air is a late 2014 model (so not a fast one).

Can you give me more details about your Safari version or the Hardware? Like screen resolution or if using a retina display.

**Reply by ScheevRapid** [reply]
Hi @matteobruni, my device is iMac 27 inch with 5k display and it looks like that this is cause behind lags. I tried to resize window to casual 1366p and lags are not present :) Thanks for an idea

**Reply by matteobruni** [reply]
I'm waiting for an Apple Silicon 16" MacBook Pro, so maybe I can test it more on retina displays.

I think the high number of particles for a 5K display could be the issue, you can set a limit if you want (or try to set some other values with the `particles.number` options)
